src/random.cpp

```diff
@@ -304,6 +311,22 @@ void GetOSRand(unsigned char *ent32)
         } else {
             RandFailure();
         }
+    } else {
+#if defined(__has_feature)
+#if __has_feature(memory_sanitizer)
```

Comment:
I don't like that we're patching our RNG to "work around" a tool. I have not tested it yet, but why couldn't we a [sanitizer special case list](https://clang.llvm.org/docs/SanitizerSpecialCaseList.html), rather than this patch? If the issue is just with `syscall`, then the list won't be very long, as this is the only place we are using it in `/src`.

Reply:
Agree with this ^

Reply:
I tried using a sanitizer special case list, https://github.com/fanquake/core-review/commit/9158e2d0d56b6aab86a8ee47eb8d8a6a70800c5a, and it seems to work ok.

Reply:
Good points. I also dislike patching non-testing code for testing purposes: I've now removed the patching.

Since MemorySanitizer (MSAN) does not support tracking memory initialization done by using the Linux `getrandom` I'm now opting out of `getrandom` usage by undefining `HAVE_SYS_GETRANDOM`.

That is done without touching anything in `src/`.